zeebe/engine/src/test/java/io/camunda/zeebe/engine/metrics/IncidentMetricsTest.java

```diff
@@ -0,0 +1,123 @@
+
+public class IncidentMetricsTest {
+
+  private static final String PROCESS_ID = "process";
+
+  @Rule public final EngineRule engine = EngineRule.singlePartition();
+  @Rule public final TestWatcher recordingExporterTestWatcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldUpdateIncidentMetricsIfIncidentOccurs() {
+    // given - process with a call activity referencing a non-existing process
+    engine
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .callActivity("call", a -> a.zeebeProcessId("non-existing"))
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when - create process instance that will cause an incident
+    final long processInstanceKey = engine.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then - wait for incident to be created and verify metrics
+    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
+    assertIncidentMetrics(
+        1.0, // created
+        1.0, // pending
+        0.0); // resolved
+  }
+
+  @Test
+  public void shouldUpdateIncidentMetricsIfIncidentIsResolved() {
+    // given - process with a call activity referencing a non-existing process
+    engine
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .callActivity("call", a -> a.zeebeProcessId("non-existing"))
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey = engine.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
+
+    // when - cancel the process instance, resolving the incident
+    engine.processInstance().withInstanceKey(processInstanceKey).cancel();
+
+    // then - wait for process to be canceled and verify metrics
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_TERMINATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
+
+    // Assert that incident created counter was incremented
+    assertIncidentMetrics(1.0, 0.0, 1.0);
+  }
+
+  private void assertIncidentMetrics(
+      final Double expectedCreated, final Double expectedPending, final Double expectedResolved) {
+    assertThat(createdIncidentsMetric())
+        .describedAs("Created incidents metric")
+        .isNotNull()
+        .isEqualTo(expectedCreated);
+    assertThat(pendingIncidentsMetric())
+        .describedAs("Pending incidents metric")
+        .isNotNull()
+        .isEqualTo(expectedPending);
+    assertThat(resolvedIncidentsMetric())
+        .describedAs("Resolved incidents metric")
+        .isNotNull()
+        .isEqualTo(expectedResolved);
```

Comment:
🔧 There's no need for an explicit `isNotNull` check because the `isEqualTo` already verifies this implicitly.